addons/mrp/models/mrp_bom.py

```diff
@@ -108,6 +108,18 @@ def _check_bom_lines(self):
                             bom_product=bom_line.parent_product_tmpl_id.display_name
                         ))
 
+    @api.onchange('bom_line_ids', 'product_qty')
+    def onchange_bom_structure(self):
+        if self.type == 'phantom' and self._origin and self.env['stock.move'].search([('bom_line_id', 'in', self._origin.bom_line_ids.ids)], limit=1):
```

Comment:
Hi @adwid 
maybe add sudo(), to be sur the check is made on all sm.
maybe add ('state','!=','cancel'), to prevent a wrong message if there are only cancelled moves.
maybe add an index on bom_line_id (for database with lot of sm).